File: python/monitor.py
import os
import pymongo.monitoring


class CommandMonitor(pymongo.monitoring.CommandListener):
    def started(self, event):
        print(f">> {event.command_name} {event.command}")

    def succeeded(self, event):
        print(f"<< {event.command_name} {event.reply}")

    def failed(self, event):
        pass


def setup_monitoring():
    if os.environ.get("PRINT_MONGODB_COMMANDS"):
        pymongo.monitoring.register(CommandMonitor())

File: ruby/monitor.rb
class CommandMonitor
  def started(event)
    puts ">> #{event.command_name} #{event.command}"
  end

  def succeeded(event)
    puts "<< #{event.command_name} #{event.reply}"
  end

  def failed(event); end
end

def setup_monitoring(client)
  return unless ENV["PRINT_MONGODB_COMMANDS"]
  client.subscribe(Mongo::Monitoring::COMMAND, CommandMonitor.new)
end
